resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -931,4 +934,12 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
             logger.warn("{} Unexpected exception: ", ch, cause);
         }
     }
+
+    @Override
+    protected InetSocketAddressResolver newInetSocketAddressResolver() {
+        HostsFileNameResolver hostsFileNameResolver = new HostsFileNameResolver(executor());
+        CompositeNameResolver<InetAddress> compositeResolver =
+                new CompositeNameResolver<InetAddress>(executor(), hostsFileNameResolver, this);
+        return new InetSocketAddressResolver(executor(), compositeResolver);
```

Comment:
This doesn't look right to me, because `DnsNameResolver` itself doesn't handle `/etc/hosts` file and its `AddressResolver` does. For consistency, we could:

- Incorporate `/etc/hosts` handling into `DnsNameResolver` or
- Add factory classes such as `NameResolvers` and `AddressResolvers` that provide some useful factory methods in `netty-resolvers`. The factory methods could use reflection to detect `netty-resolver-dns` and instantiate the composite resolver.
- or you could suggest something better.